includes/class-wc-amazon-payments-advanced-rest-api-controller.php

```diff
@@ -583,8 +577,8 @@ public function refund( $request ) {
 	 * @return null|WP_Error Null if there's no error in the request.
 	 */
 	protected function get_missing_reference_id_request_error( $order_post ) {
-		$ref_id = get_post_meta( $order_post->ID, 'amazon_reference_id', true );
-		if ( ! $ref_id ) {
+		$reference_id = $order_post->get_meta( 'amazon_reference_id', true, 'edit' );
```

Comment:
It looks like the doc-block for `$order_post` is outdated, + we probably could give it a better name. WDYT?

Reply:
done!